r/R/utilities_expression.R

```diff
@@ -0,0 +1,62 @@
+#' Returns expression values for selected genes
+#'
+#' @param genes - Must have names and input(ensmbl ids)
+#' @param data
+#'
+#' @return
+#' @export
+#'
+#' @examples
+getExpressionValues <- function(genes, data) {
+  quantile_threshold <- 0.95
+  #
+  # Get the expression values for those genes in the corresponding matrix.
+  geneExpression <- data@assays$RNA@data[unique(genes$input), , drop = FALSE]
+  geneExpression <- as.data.frame(t(as.matrix(geneExpression)))
+  geneExpression$cells_id <- data@meta.data$cells_id
+  geneExpression <- geneExpression[order(geneExpression$cells_id), ]
+  geneExpression <- geneExpression %>%
+    tidyr::complete(cells_id = seq(0, max(data@meta.data$cells_id))) %>%
+    dplyr::select(-cells_id) %>%
+    as.data.frame()
+  # worried about duplicate gene row.names in @data
+  symbol_idx <- match(colnames(geneExpression), genes$input)
+  colnames(geneExpression) <- genes$name[symbol_idx]
+  adjGeneExpression <- as.data.frame(apply(geneExpression, 2, FUN = function(x) {
```

Comment:
I see you are trying to adjust Gene expression in some way. Can you explain what are you adjusting in particular?

Reply:
This is used to generate the truncated gene expression, where we truncate values to a 95th percentile in order to improve visualization